### PR TITLE
Compatibility to TYPO3 v12

### PR DESCRIPTION
**Pull request by sgrossberndt**

### CODE REVIEW

File: Classes/EventListener/ModifyFileDumpEventListener.php [hunk 1]
```diff
@@ -205,14 +170,13 @@ public function checkFileAccess(ResourceInterface $file)
...
-     * @todo: find a nicer way to force the download. Other hooks are blocked by this.
-     * @todo: Try to get the resumable option part of TYPO3 core itself
+     * TODO: Try to get the resumable option part of TYPO3 core itself find a nicer way to force the download. Other hooks are blocked by this.
      *
      * @param File $file
      * @param bool $asDownload
      * @param bool $resumableDownload
      */
-    protected function dumpFileContents($file, $asDownload, $resumableDownload)
+    protected function dumpFileContents(File $file, bool $asDownload, bool $resumableDownload)
```
Comment (FamousWolf) [inline Classes/EventListener/ModifyFileDumpEventListener.php:179]: `$file` can also be `ProcessedFile` (this was already wrong in the annotation, but that didn't cause any problems). `File` can be replaced with `FileInterface`

File: Classes/Security/CheckPermissions.php [hunk 2]
```diff
@@ -301,7 +277,7 @@ public function matchFeGroupsWithFeUser($groups, $userFeGroups)
         }
 
         // enabled for all loggedIn Users
-        if (str_contains($groups, '-2')) {
+        if (strpos($groups, '-2') !== false) {
```
Comment (FamousWolf) [inline Classes/Security/CheckPermissions.php:280]: This is not needed, as TYPO3 11 uses symfony polyfill for PHP 8.0 functions. So you can use str_contains, even if using PHP 7.4. This is also used in the core (see https://forge.typo3.org/issues/95466)

**Review by FamousWolf — COMMENTED** [review]
Looks good. I found 2 small things.

**Comment by sgrossberndt** [comment]
Addressed both issues.

**Pull request merged**